components/compliance-service/compliance.go

```diff
@@ -195,11 +199,15 @@ func serveGrpc(ctx context.Context, db *pgdb.DB, connFactory *secureconn.Factory
 	version.RegisterVersionServiceServer(s, versionserver.New())
 	status.RegisterComplianceStatusServer(s, statusSrv)
 
-	purgeServer, err := setupDataLifecyclePurgeInterface(ctx, connFactory, conf, cerealManager)
-	if err != nil {
-		logrus.Fatalf("serveGrpc aborting, can't setup purge server: %s", err)
+	if os.Getenv("RUN_MODE") == "test" {
+		logrus.Warn(`Skipping data-lifecycle setup due to RUN_MODE env var being set to "test"`)
+	} else {
+		purgeServer, err := setupDataLifecyclePurgeInterface(ctx, connFactory, conf, cerealManager)
```

Comment:
I thought we already handled it for data lifecycle in the setup function https://github.com/chef/automate/blob/5354eae026acae85e257cb9e18a70834bf57d67a/components/compliance-service/compliance.go#L443

Reply:
I don't think that will stop the manager this creates from polling for jobs every few seconds and creating an error trying to talk to cereal-service

Reply:
Cool cool, I think we should probably combine those. It doesn't make sense to check for test in a function that won't be called.